cajita/src/Cajita_Array.hpp

```diff
@@ -471,31 +545,59 @@ void update( Array_t& a, const typename Array_t::value_type alpha,
         } );
 }
 
+template <class Array_t, class DecompositionTag>
+std::enable_if_t<2 == Array_t::num_space_dim, void>
+update( Array_t& a, const typename Array_t::value_type alpha, const Array_t& b,
+        const typename Array_t::value_type beta, DecompositionTag tag )
+{
+    static_assert( is_array<Array_t>::value, "Cajita::Array required" );
```

Comment:
Do we assume that Array a and b are made from the same layout? do we need to check their extent of indexspace?

Reply:
`a` and `b` are both `Array_t` and hence this won't compile if they are of the same type and have the same layout. I do agree though that they could technically have different sizes so we should check that. Let's make a new PR for that as this PR is just to add 2D/3D